### ISSUE TITLE
Add an option to skip SSL certificate validation when connecting to SSL-aware mail server

### ISSUE THREAD

**Issue by lospejos** [bbottema/simple-java-mail#48]
Would be nice to add an option to skip SSL certificate validation when connecting to SSL-aware mail server which has selfsigned or unvalidatable certificate

**Comment by lospejos** [comment]
Look here: 
http://stackoverflow.com/questions/4894954/how-to-ignore-server-cert-error-in-javamail
http://www.rgagnon.com/javadetails/java-fix-certificate-problem-in-HTTPS.html

**Comment by bbottema** [comment]
I will add it, but just so you know you can already add it:

```
Properties props = new Properties();
props.setProperty("mail.smtp.ssl.trust", "*");
mailer.applyProperties(props);
```
Or:
```
mailer.getSession().getProperties().put("mail.smtp.ssl.trust", "*");
```

**Comment by bbottema** [comment]
With the next release you can set the property programmatically, like so:

mailer.trustAllSSLHosts(true);
mailer.trustAllSSLHosts(false);

mailer.trustSSLHosts("a", "b", "c", ...);

**Comment by bbottema** [comment]
Released in v4.1.0.

**Comment by bbottema** [comment]
Bug #110 was fixed for this feature in 5.0.0.rc1-SNAPSHOT.

**Comment by bbottema** [comment]
Historical follow-up: #677 keeps this trust-bypass capability as an explicit compatibility escape hatch, but Simple Java Mail 9.2.0 no longer enables wildcard certificate trust by default. Normal connections now use the JVM trust store while retaining hostname verification.
